.github/workflows/ci.yml

```diff
@@ -1,96 +1,83 @@
-          cpanm -n Dist::Zilla
-          dzil authordeps --missing | cpanm -n
-      - name: Make distribution
-        shell: bash
-        run: |
-          eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
-          dzil build --in build-dir
-      - name: Upload artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: dist
-          path: ./build-dir
-  test:
-    needs: dist
+      - name: 'build-dzil-dist'
+        uses: PDLPorters/devops/github-actions/build-dzil-dist@master
+  ci:
     runs-on: ${{ matrix.os }}
+    if: ${{ github.event_name == 'push' || github.event_name == 'pull_request' }}
+    needs: [ notify, dist ]
     strategy:
       fail-fast: false
       matrix:
         os: [macos-11, windows-latest, ubuntu-latest]
-        perl: ['5']
+        perl-version: ['5']
         include:
-          - { os: 'ubuntu-latest', perl: "5.14" }
-          - { os: 'ubuntu-latest', perl: "5.16" }
-          - { os: 'ubuntu-latest', perl: "5.20" }
-          - { os: 'ubuntu-latest', perl: "5.30" }
-          - { os: 'ubuntu-latest', perl: "5.32" }
-    name: Perl ${{ matrix.perl }} on ${{ matrix.os }}
-
+          - { os: 'ubuntu-latest', perl-version: "5.14" }
+          - { os: 'ubuntu-latest', perl-version: "5.16" }
+          - { os: 'ubuntu-latest', perl-version: "5.20" }
+          - { os: 'ubuntu-latest', perl-version: "5.30" }
+          - { os: 'ubuntu-latest', perl-version: "5.32" }
     steps:
+      - uses: actions/checkout@v2
       - name: Get dist artifact
         uses: actions/download-artifact@v2
         with:
           name: dist
-
-      - name: Set up perl
-        uses: shogo82148/actions-setup-perl@v1
-        if: matrix.os != 'windows-latest'
+          path: build-dir
+      - name: 'ci-dist: target-setup-perl'
+        uses: PDLPorters/devops/github-actions/ci-dist@master
         with:
-          perl-version: ${{ matrix.perl }}
-      - name: Set up perl (Strawberry)
-        uses: shogo82148/actions-setup-perl@v1
-        if: matrix.os == 'windows-latest'
+          path: build-dir
+          target-setup-perl: true
+          perl-version: ${{ matrix.perl-version }}
+      - name: 'ci-dist: target-all after install-dist-perl-deps'
+        uses: PDLPorters/devops/github-actions/ci-dist@master
         with:
-          distribution: 'strawberry'
-
-      - name: target-setup-perl (perl -V)
-        shell: bash
-        run: |
-          echo "::group::setup-perl (perl -V)"
-          export MYPERL=$(which -a perl | grep -m 1 hostedtoolcache)
-          echo "MYPERL=$MYPERL" >> $GITHUB_ENV
-          $MYPERL -V
-          echo "::endgroup::"
+          path: build-dir
+          target-setup-perl: false
+          target-install-dist-perl-deps: true
+          target-test-release-testing: true
+          target-test: true
+          test-enable-release-testing: ${{ matrix.release-test }}
+          test-enable-coverage: ${{ matrix.coverage }}
```

Comment:
Do these mean we need to add jobs for each of these?

Reply:
Yes.

I believe that Coveralls needs to be enabled for this repo. I can’t as I don’t have permissions.

Reply:
I think it's already enabled? https://coveralls.io/github/ebaudrez/Test-PDL

Reply:
Great! It should pass with coverage now with this change.